internal/dao/helpers.go

```diff
@@ -81,3 +84,16 @@ func ToYAML(o runtime.Object, showManaged bool) (string, error) {
 
 	return buff.String(), nil
 }
+
+// serviceAccountMatches validates that the ServiceAccount referenced in the PodSpec matches the incoming
+// ServiceAccount. If the PodSpec ServiceAccount is blank kubernetes will use the "default" ServiceAccount
+// when deploying the pod, so if the incoming SA is "default" and PodSpec is an empty string that is also a match.
+func serviceAccountMatches(podSpec *v1.PodSpec, saName string) bool {
+	if podSpec.ServiceAccountName != saName && podSpec.ServiceAccountName != "" {
```

Comment:
I think this could be simplified a bit?? ie something like..
```go
func saMatch(rsa, sa string) bool {
   if rsa == "" {
    rsa = defaultServiceAcct
   }
   
   return rsa == sa
}
```

Reply:
Agreed, updated.